src/views/ipv/page/pyi-update-details.njk

```diff
@@ -0,0 +1,122 @@
+{% extends "shared/base.njk" %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/checkboxes/macro.njk" import govukCheckboxes %}
+{% from "govuk/components/summary-list/macro.njk" import govukSummaryList %}
+
+{% set pageTitleKey = 'pages.pyiUpdateDetails.title' %}
+{% set googleTagManagerPageId = "pyiUpdateDetails" %}
+
+{% set errorState = pageErrorState %}
+{% set errorTitle = 'pages.pyiUpdateDetails.content.formErrorMessage.errorSummaryTitleText' | translate %}
+{% set errorText = 'pages.pyiUpdateDetails.content.formErrorMessage.errorSummaryDescriptionText' | translate %}
+
+{% block content %}
+    <h1 class="govuk-heading-l" id="header" data-page="{{ googleTagManagerPageId }}"
+        xmlns="http://www.w3.org/1999/html">{{ 'pages.pyiUpdateDetails.header' | translate }}</h1>
+
+    {% set givenNames = [] %}
+    {% for namePart in userDetails.nameParts %}
+      {% if namePart.type == "GivenName" %}
+        {% set givenNames = (givenNames.push(namePart.value), givenNames) %}
+      {% endif %}
+    {% endfor %}
+
+    {% set familyNames = [] %}
+    {% for namePart in userDetails.nameParts %}
+      {% if namePart.type == "FamilyName" %}
+        {% set familyNames = (familyNames.push(namePart.value), familyNames) %}
+      {% endif %}
+    {% endfor %}
```

Comment:
I'd probably slightly prefer this logic to be done in `reuseHelper.js` - it'll be easier than here, and it'll be reusable if we need to display the given/family name on another page.

Not critical - we could wait until we know we need to reuse it.

Reply:
Makes sense. Didn't think of it.